### PR TITLE
Ordertype

### PR DESCRIPTION
**Pull request by TomaszGol**

### CODE REVIEW

File: src/eventprocessor/market/orderCreatedEventProcessor.ts [hunk 1]
```diff
@@ -17,31 +18,59 @@ export class OrderCreatedEventProcessor implements EventProcessor {
...
-        quantity: BigInt(quantity),
-        initialQuantity: BigInt(quantity),
-        who: encodeUserValue(parsedEvent.who),
-        blockHeight: BigInt(block.header.height),
-        // @ts-ignore
-        timestamp: new Date(block.header.timestamp),
-        status: OrderStatus.ACTIVE,
-      });
+      let order: Order;
+      if (call.name === this.callName) {
```
Comment (Klapeyron) [inline src/eventprocessor/market/orderCreatedEventProcessor.ts:36]:
```suggestion
      if (call.name === "Market.create_order") {
```

File: src/main.ts [hunk 2]
```diff
@@ -13,8 +13,10 @@ processor.run(
     for (let block of ctx.blocks) {
       await CheckIfMarketClosed.closeMarket(ctx, block);
       for (let event of block.events) {
-        let processor = processorProvider.getProcessorByName(event.name);
-        await processor?.process(ctx, block, event);
+        for (let call of block.calls) {
```
Comment (Klapeyron) [inline src/main.ts:16]: it should not be nested probably, due event is a subelement of call

**Review by Klapeyron — CHANGES_REQUESTED** [review]

File: src/eventprocessor/market/orderCreatedEventProcessor.ts (hunk 1 above)
Comment (TomaszGol) [reply src/eventprocessor/market/orderCreatedEventProcessor.ts:36]: "Market.create_order" is this.callName, I think using const here can be better solution

File: src/main.ts (hunk 2 above)
Comment (TomaszGol) [reply src/main.ts:16]: block.calls is array, so we need to somehow loop through it to get call. If event is subelement of call, we will not catch same event 2 times.

**Review by Klapeyron — APPROVED** [review]

**Pull request merged**